cloudinit/config/cc_ansible.py

```diff
@@ -99,32 +103,39 @@ def install(self, pkg_name: str):
 
 class AnsiblePullPip(AnsiblePull):
     def __init__(self, distro: Distro):
-        self.cmd_pull = ["ansible-pull"]
-        self.cmd_version = ["ansible-pull", "--version"]
-        self.env["PATH"] = ":".join([self.env["PATH"], "/root/.local/bin/"])
-        self.distro = distro
+        super().__init__(distro)
+
+        ansible_path = "/root/.local/bin/"
+        old_path = self.env.get("PATH")
+        if old_path:
+            self.env["PATH"] = ":".join([old_path, ansible_path])
+        else:
+            self.env["PATH"] = ansible_path
 
     def install(self, pkg_name: str):
         """should cloud-init grow an interface for non-distro package
         managers? this seems reusable
         """
         if not self.is_installed():
             # bootstrap pip if required
-            if not which("pip3"):
-                self.distro.install_packages(PIP_PKG)
-            subp(["python3", "-m", "pip", "install", "--user", pkg_name])
+            try:
+                import pip
+            except ImportError:
+                self.distro.install_packages(self.distro.pip_package_name)
+
+            self.subp(
+                [sys.executable, "-m", "pip", "install", "--user", pkg_name]
```

Comment:
cloud-init is running as root.
how sensible is `--user` here?

Reply:
Hey @igalic. Interesting question. It actually raises a couple of other questions that I'll get to in a moment.

To directly answer the question:
```
  --user                      Install to the Python user install directory for
                              your platform. Typically ~/.local/, or
                              %APPDATA%\Python on Windows. (See the Python
                              documentation for site.USER_BASE for full
                              details.)
```

I don't think it necessarily hurts to install in `/root/.local`, but also doesn't really add anything of value either.

I'm fine with dropping it when installing as root, however as you said:

> cloud-init is running as root.

And installing with pip as an elevated user is generally not considered best practice. This begets change to support installing as non-root.

To enable users to install and run ansible without running pip as root (and also to run ansible as an unprivileged user if they wish), I'd like to add a couple of keys to this module: `install-user`, and `run-user`. This would allow cloud-init to drop permissions and run as a user already configured via cloud-init's `cc_users_groups` module.

When running as an unprivileged user, I think that the `--user` flag makes more sense and should be used in this case.

#### To Summarize

I'll drop the `--user` flag here. In a follow-up PR, I'll add some new keys that allow installing and running ansible as alternate users to support best practices.

@igalic - Thoughts?